### PR TITLE
Add skipEmptyLines option to ParseOptions and parsing functions.

### LINKED ISSUE #516: Add skipEmptyLines option to ignore blank rows in CSV
## Description
When parsing CSV files, blank/empty lines are currently processed as records with empty fields. Many CSV parsers provide a `skipEmptyLines` option to ignore these blank rows, which is useful for handling real-world CSV files.

## Current Behavior
```typescript
const csv = `name,age
Alice,42

Bob,69`;

const records = await parse.toArray(csv);
// Result: [{ name: 'Alice', age: '42' }, { name: '', age: '' }, { name: 'Bob', age: '69' }]
```

The empty line creates a record with empty string values.

## Proposed Behavior
```typescript
const csv = `name,age
Alice,42

Bob,69`;

const records = await parse.toArray(csv, { skipEmptyLines: true });
// Result: [{ name: 'Alice', age: '42' }, { name: 'Bob', age: '69' }]
```

### 1. Add option to types
In `src/common/types.ts`, add to `ParseOptions`:
```typescript
export interface ParseOptions<...> {
  // ... existing options
  
  /**
   * Skip empty lines in CSV.
   * An empty line is a line with no content or only whitespace between delimiters.
   * @default false
   */
  skipEmptyLines?: boolean;
}
```

### 2. Modify RecordAssembler
In `src/RecordAssembler.ts`, add logic to skip records where all fields are empty:
- Check if all values in `#row` are empty strings
- If `skipEmptyLines` is true and all fields are empty, skip yielding the record

### 3. Update all parse functions
Pass the `skipEmptyLines` option through:
- `parseString()`
- `parseStringStream()`
- `parseBinary()`
- etc.

### 4. Add tests
Add test cases in `src/parse.spec.ts`:
- CSV with blank lines at various positions (beginning, middle, end)
- CSV with lines containing only spaces/tabs between delimiters
- Property-based test with random blank line insertion

## Real-World Use Case
CSV exports from Excel or databases often contain blank rows for formatting. This feature makes parsing more robust.

## References
- Papa Parse: `skipEmptyLines` option
- csv-parse (Node.js): `skip_empty_lines` option
- Python csv module: similar behavior with DictReader

## Acceptance Criteria
- [ ] Add `skipEmptyLines` option to `ParseOptions`
- [ ] Implement logic in `RecordAssembler`
- [ ] Update all parsing functions to support the option
- [ ] Add comprehensive tests
- [ ] Update TypeDoc comments
- [ ] Default value should be `false` for backward compatibility

---
**Good First Issue**: Clear specification, touches core parsing logic but well-scoped!

### PR DESCRIPTION
**Pull request by VaishnaviOnPC**
Extended ParseOptions to include a skipEmptyLines?: boolean flag.
Updated the parser logic to skip over empty rows when this option is enabled.
Added comprehensive property-based tests in parse.spec.ts to verify correct behavior with and without this option.

Closes #516

## Summary by CodeRabbit

* **New Features**
  * Added an optional skipEmptyLines setting for CSV parsing. When enabled, fully empty lines are omitted from parsed output, preventing spurious empty records (defaults to off).

* **Tests**
  * Added tests validating skipEmptyLines across varied CSV inputs, ensuring it does not increase record counts and tolerates expected parse edge cases.

### CODE REVIEW

File: src/RecordAssembler.ts [hunk 1]
```diff
@@ -55,6 +59,17 @@ export class RecordAssembler<Header extends ReadonlyArray<string>> {
           if (this.#header === undefined) {
             this.#setHeader(this.#row as unknown as Header);
           } else {
+            const isEmptyLine =
```
[the remark this answers is not in the document]
Comment (VaishnaviOnPC) [reply src/RecordAssembler.ts]: Yes, it works. I will make that change.

File: src/RecordAssembler.ts [hunk 2]
```diff
@@ -18,8 +18,11 @@ export class RecordAssembler<Header extends ReadonlyArray<string>> {
   #dirty = false;
   #signal?: AbortSignal;
   #maxFieldCount: number;
+  #skipEmptyLines: boolean;
 
-  constructor(options: RecordAssemblerOptions<Header> = {}) {
+  constructor(
+    options: RecordAssemblerOptions<Header> & { skipEmptyLines?: boolean } = {},
```
Comment (kamiazya) [inline src/RecordAssembler.ts]: ​Since it's no longer necessary to extend the type here, please remove this.

File: src/RecordAssembler.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (kamiazya) [reply src/RecordAssembler.ts]: Thank you. But with this approach, the dirty validation will be performed twice. I thought it would be more efficient to check it in the else block of the if (dirty) statement, like this:

````
if #dirty
    yield CSVRecord
else
    if skipEmptyLines
        continue
    else
       yield Empty CSVRecord
````

**Review by kamiazya — APPROVED** [review]
LGTM!
